docs/asciidoc/faq.adoc

```diff
@@ -205,5 +226,113 @@ processor.publishOn(scheduler1)
          .doOnNext(i -> processNext(i))
          .subscribe();
 ----
+====
+
+== What Is a Good Pattern for Contextual Logging? (MDC)
+As described in section <<context>>, `ThreadLocal` does not make much sense in Reactive Streams, where several streams could run steps on a common thread.
+Instead of using `ThreadLocal`, Reactor offers the `Context`, a map-like object that is tied to a `Subscription` and not a `Thread`.
+
+One recurring use of `ThreadLocal` in Java is for logging frameworks to store variables that are used in their logging pattern, generally by way of a `Map` called the MDC ("Mapped Diagnostic Context").
+
+Unfortunately, this pattern is hard to opt out of and hard to apply transparently from a reactive library perspective.
+This entry of the FAQ offers a possible intermediate solution when one wants to perform log statements in relation to events in the stream (`onNext`, `onError`, and `onComplete`).
+
+The most straightforward way to perform such logging and get contextual information from the `Context` to the MDC is to wrap logging statements in a `doOnEach` operator with a little bit of boilerplate code.
+This boilerplate depends on both the logging framework/abstraction of your choice and the information you want to put in the MDC, so it has to be in your codebase.
+
```

Comment:
One thing that might puzzle someone, who has not used the context, is how that context is inserted in the first place. Perhaps worth pointing out that the context key needs to be inserted early in the execution chain, possibly through a `WebFilter` (e.g. in a WebFlux application) so that it is available throughout. Or if there is a convenient place to link to for learning more on how to insert context.

Reply:
added a small hint at that with a more explicit recommendation to read the `<<Context>>` section